core-container/entry.sh

```diff
@@ -16,88 +16,487 @@
+        if kubectl get ingress kubestellar-ingress &> /dev/null; then
+            EXTERNAL_HOSTNAME=$(kubectl get ingress kubestellar-ingress -o yaml -o jsonpath={.spec.rules[0].host} 2> /dev/null)
+        fi
     fi
-fi
+    echo "${EXTERNAL_HOSTNAME}"
+}
 
-# Create the certificates
-if [ -n "$EXTERNAL_HOSTNAME" ]; then
-    echo "< Ensuring the TLS certificate >---------------------------"
-    eval pieces=($(kubestellar-ensure-kcp-server-creds ${EXTERNAL_HOSTNAME}))
-    echo "TLS certificates for server ${EXTERNAL_HOSTNAME}:"
-    echo ${pieces[0]}
-    echo ${pieces[1]}
-    echo ${pieces[2]}
-fi
 
-# Start kcp
-echo "< Starting kcp >-------------------------------------------"
+function create_or_replace() { # usage: filename
+    filename="$1"
+    kind=$(grep kind: "$filename" | head -1 | awk '{ print $2 }')
+    name=$(grep name: "$filename" | head -1 | awk '{ print $2 }')
+    if kubectl get "$kind" "$name" &> /dev/null ; then
+        kubectl replace -f "$filename"
+    else
+        kubectl create -f "$filename"
+    fi
+}
 
-mkdir -p ${PWD}/kubestellar-logs # required in Open Shift clusters
+# function ensure_root_compute_configd() {
+#     echo "Ensuring configd for root:compute..."
 
-if [ -n "$EXTERNAL_HOSTNAME" ]; then
-    echo -n "Running kcp with TLS keys... "
-    kcp start --tls-sni-cert-key ${pieces[1]},${pieces[2]} &>> "${PWD}/kubestellar-logs/kcp.log" &
-else
-    echo -n "Running kcp without TLS keys... "
-    kcp start &>> "${PWD}/kubestellar-logs/kcp.log" &
-fi
-echo "logfile=./kubestellar-logs/kcp.log"
+#     if ! kubectl ws root:compute &> /dev/null ; then
+#         echoerr "something is very wrong, unable to set current kcp workspace to root:compute"
+#         exit 1
+#     fi
 
-echo "Waiting for kcp to be ready... it may take a while"
-until [ "$(kubectl ws root:compute 2> /dev/null)" != "" ]; do
-    sleep 5
-done
+#     ( cd /home/kubestellar/config/kube/exports/namespaced
+#         # Some are too big to `kubectl apply`
+#         for rsfn in apiresourceschema-*.yaml; do
+#             create_or_replace $rsfn
+#         done
+#         for refn in apiexport-*.yaml; do
+#             kubectl apply -f "$refn"
+#         done
+#     )
 
-echo "kcp version: $(kubectl version --short 2> /dev/null | grep kcp | sed 's/.*kcp-//')"
+#     kubectl apply -f - <<EOF
+# apiVersion: rbac.authorization.k8s.io/v1
+# kind: ClusterRole
+# metadata:
+#   name: compute:apiexport:kubernetes-extended:bind
+# rules:
+# - apiGroups:
+#   - apis.kcp.io
+#   resourceNames:
+#   - apps
+#   - autoscaling
+#   - batch
+#   - core.k8s.io
+#   - discovery.k8s.io
+#   - networking.k8s.io
+#   - policy
+#   - storage.k8s.io
+#   resources:
+#   - apiexports
+#   verbs:
+#   - bind
+# ---
+# apiVersion: rbac.authorization.k8s.io/v1
+# kind: ClusterRole
+# metadata:
+#   name: compute:apiexport:kubernetes:maximal-permission-policy-extended
+# rules:
+# - apiGroups:
+#   - ""
+#   - apps
+#   - networking.k8s.io
```

Comment:
Not loving a PR that introduces a bunch of commented out code.

Reply:
Removed, I commented it during test.
New code is working.